@types/react-native don't work with typescript 3.1.1

- [x] I tried using the latest stable version of tsc (`typescript@latest` = `typescript@3.1.1`). https://www.npmjs.com/package/typescript
- [x] I tried using the `@types/react-native@0.57.0` package and had problems.
- [x] [Mention](https://github.com/blog/821-mention-somebody-they-re-notified) the authors (see `Definitions by:` in `index.d.ts`) so they can respond.
  - Authors: @alloy, @huhuanming, @iRoachie, @skn0tt, @timwangdev, @kamal, @nelyousfi, @alexdunne, @swissmanu, @bm-software, @tkrotoff, @a-tarasyuk, @mvdam, @esemesek

Current typings produce very strange errors being used with typescript 3.1.1
I submit https://github.com/Microsoft/TypeScript/issues/27421 to typescript repo, but sure that DT also should track this issue. Main problem, that I can't create minimal example for this. And I can't personally understand why ts since 3.1 behave so strangely. It would be good if anyone can help with this.

Next is mostly duplication of info from https://github.com/Microsoft/TypeScript/issues/27421:

```
import React from 'react'
import { StyleSheet, Text } from 'react-native'

const s = StyleSheet.create({
  didNotWork: {
    fontSize: 16,
    fontWeight: '900', // if we comment this line, errors gone
    marginTop: 5, // if this line commented, errors also gone
  },
  work: {
    fontSize: 18,
//    fontWeight: '900', // when uncommented also work
  },
})

export const sample1 = <Text style={s.work} />
export const sample2 = <Text style={s.didNotWork} />
// ^ this line generate error:
// index.tsx:17:30 - error TS2322: Type 'RegisteredStyle<{ fontSize: number; fontWeight: string; marginTop: number; }>' is not assignable to type 'StyleProp<TextStyle>'.
//   Type 'RegisteredStyle<{ fontSize: number; fontWeight: string; marginTop: number; }>' is not assignable to type 'RecursiveArray<false | TextStyle | RegisteredStyle<TextStyle> | null | undefined>'.
//     Property 'length' is missing in type 'Number & { __registeredStyleBrand: { fontSize: number; fontWeight: string; marginTop: number; }; }'.
// 17 export const sample2 = <Text style={s.didNotWork} />
//                                 ~~~~~
//   node_modules/@types/react-native/index.d.ts:907:5
//     907     style?: StyleProp<TextStyle>;
//             ~~~~~
//     The expected type comes from property 'style' which is declared here on type 'IntrinsicAttributes & IntrinsicClassAttributes<Text> & Readonly<{ children?: ReactNode; }> & Readonly<TextProps>'
export const sample3 = <Text style={{fontSize: 16, fontWeight: '900', marginTop: 5}} />
```

This sample shows the error. There is sandbox repo: https://github.com/vovkasm/rn-ts-3.1.1 (to reproduce, clone and `npm install && npm test`)
1. If I comment out `fontWeight` or `marginTop` style of `didNotWork` key, error dissapeared
2. If I add `fontWeight` to `work` key, error dissapeared

Why is this?

--- a/types/react-native/index.d.ts
+++ b/types/react-native/index.d.ts
@@ -5226,7 +5226,7 @@ export namespace StyleSheet {
     /**
      * Creates a StyleSheet style reference from the given object.
      */
-    export function create<T extends NamedStyles<T>>(styles: T): { [P in keyof T]: RegisteredStyle<T[P]> };
+    export function create<T extends NamedStyles<T> | NamedStyles<any>>(styles: T): { [P in keyof T]: RegisteredStyle<T[P]> };
 
     /**
      * Flattens an array of style objects, into one aggregated style object.

--- a/types/react-native/test/index.tsx
+++ b/types/react-native/test/index.tsx
@@ -14,7 +14,6 @@ For a list of complete Typescript examples: check https://github.com/bgrieder/RN
 import * as PropTypes from "prop-types";
 import * as React from "react";
 import {
-    RegisteredStyle,
     Alert,
     AppState,
     AppStateIOS,
@@ -56,6 +55,7 @@ import {
     SectionListRenderItemInfo,
     Switch,
     RefreshControl,
+    RegisteredStyle,
     TabBarIOS,
     NativeModules,
     MaskedViewIOS,

--- a/types/react-native/test/index.tsx
+++ b/types/react-native/test/index.tsx
@@ -14,6 +14,7 @@ For a list of complete Typescript examples: check https://github.com/bgrieder/RN
 import * as PropTypes from "prop-types";
 import * as React from "react";
 import {
+    RegisteredStyle,
     Alert,
     AppState,
     AppStateIOS,
@@ -174,6 +175,14 @@ const imageStyle: StyleProp<ImageStyle> = {
 const viewProperty = StyleSheet.flatten(viewStyle).backgroundColor;
 const textProperty = StyleSheet.flatten(textStyle).fontSize;
 const imageProperty = StyleSheet.flatten(imageStyle).resizeMode;
+
+const s = StyleSheet.create({
+  shouldWork: {
+    fontWeight: '900', // if we comment this line, errors gone
+    marginTop: 5, // if this line commented, errors also gone
+  },
+})
+const f1: RegisteredStyle<TextStyle> = s.shouldWork
 
 const testNativeSyntheticEvent = <T extends {}>(e: NativeSyntheticEvent<T>): void => {
     e.isDefaultPrevented();

